[1.21.3] Add RenderPlayerEvent test and docs fixup

dhyces opened the pull request:
Closes #1266

Retargeting 1.21.3, this PR is now just the test and docs fixup, as the event was made to subclass `RenderLivingEvent` in the 1.21.2 port. The docs fixup affects the `RenderLivingEvent` class and the `RenderPlayerEvent` class. The test simply renders an iron block above the players head with the post event.

dhyces committed:
Make RenderPlayerEvent subclass RenderLivingEvent

TelepathicGrunt commented:
Putting a 1.21.1 tag on this PR. It is a bit too late now to move events around. We can revisit this PR on next breaking mc version. Or if you adjust this PR to only be javadoc cleanup, we can merge that and have an issue report or something to do the event moving in future mc version

dhyces commented:
Understandable. I don't think the jdoc changes are important enough to warrant a separate PR *just* for these two classes. imo, they should either be fixed in bulk or modified when the classes are changed.

dhyces committed:
Add RenderPlayerEvent test

dhyces committed:
Update for 1.21.2 changes

Removed the renderer accessor cast override as well, in case users replace the player renderer

dhyces committed:
Apply spotless

XFactHD commented on src/main/java/net/neoforged/neoforge/client/event/RenderPlayerEvent.java:
```diff
@@ -27,19 +27,19 @@
 public abstract class RenderPlayerEvent extends RenderLivingEvent<AbstractClientPlayer, PlayerRenderState, PlayerModel> {
     @ApiStatus.Internal
-    protected RenderPlayerEvent(PlayerRenderState renderState, PlayerRenderer renderer, float partialTick, PoseStack poseStack, MultiBufferSource multiBufferSource, int packedLight) {
+    protected RenderPlayerEvent(PlayerRenderState renderState, LivingEntityRenderer<AbstractClientPlayer, PlayerRenderState, PlayerModel> renderer, float partialTick, PoseStack poseStack, MultiBufferSource multiBufferSource, int packedLight) {
```
What's the thought behind loosening this from `PlayerRenderer` to `LivingRenderer`?

dhyces replied:
Mostly just that the class doesn't have a hard restriction on it. If player's override the player renderer and extend the player model, they'd be able to reuse the event. If Neo does want to strictly require the `PlayerRenderer`, then I believe the `getRenderer` method should be overridden to provide specifically `PlayerRenderer` rather than `LivingEntityRenderer<AbstractClientPlayer, PlayerRenderState, PlayerModel>`

TelepathicGrunt replied:
I think the second way is more desired. Could you undo this change? The renderer getting can be made more specific

TelepathicGrunt replied:
Dang looks like linter got you. Cursed linters!

dhyces committed:
PlayerRenderer parameter and method override

Edited file on mobile, hopes and prayers this is correct

Matyrobbrt commented:
@neoforged/bots run applyAllFormatting

dhyces commented:
That's a very handy bot :p

neoforged-automation[bot] committed:
Update formatting

Matyrobbrt committed:
Merge branch '1.21.x' into fix/sub-render-event
